src/librustdoc/clean/types.rs

```diff
@@ -873,17 +877,24 @@ impl Attributes {
                         if let Some(ref fragment) = *fragment {
                             let url = match cache.extern_locations.get(&krate) {
                                 Some(&(_, _, ExternalLocation::Local)) => {
-                                    let depth = CURRENT_DEPTH.with(|l| l.get());
-                                    "../".repeat(depth)
+                                    if relative_to[0] == "std" {
+                                        let depth = relative_to.len() - 1;
+                                        "../".repeat(depth)
+                                    } else {
+                                        let depth = relative_to.len();
+                                        format!("{}std/", "../".repeat(depth))
+                                    }
+                                }
+                                Some(&(_, _, ExternalLocation::Remote(ref s))) => {
+                                    format!("{}{}std/", s, if s.ends_with('/') { "" } else { "/" })
```

Comment:
Hardcoding "std" here and elsewhere seems wrong. Does this work when documenting crates other than the standard library?

Reply:
This is only being used for primitives, which are always in the standard library.